plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -137,9 +133,22 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
         val left = equalityOperatorCall.arguments[0].accept(this, data)
         val right = equalityOperatorCall.arguments[1].accept(this, data)
 
+        // TODO: This is actually not how Kotlin handles Comparisons
+        // The semantics of `a == b` is `a?.equals(b) ?: (b === null)` (see https://kotlinlang.org/docs/equality.html)
+        // Since we have no equality function associated with objects yet we model it with the Viper equality directly
+        // to reason about equality at all (even though this is not equivalent)
+        val commonSuperType =
```

Comment:
Is an `isSubtype` function easier to implement than an `commonSupertype` function?  I feel like the latter may be more straightforward and extensible.